Lib/queue.py

```diff
@@ -129,15 +152,22 @@ def put(self, item, block=True, timeout=None):
         Otherwise ('block' is false), put an item on the queue if a free slot
         is immediately available, else raise the Full exception ('timeout'
         is ignored in that case).
+
+        Raises a ShutDown if shutdown_state attribute is not set to
+        `_queue_alive`.
```

Comment:
```suggestion
        Raises ShutDown if the queue has been shut down.
```

Doc-strings are for humans